gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeClient.java

```diff
@@ -0,0 +1,304 @@
+                DeleteDatabaseRequest.newBuilder().setName(name.toString()).build());
+            return Empty.getDefaultInstance();
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchNamespaceException(
+                e, "Namespace does not exist: %s", name.getDatabase());
+          }
+        },
+        name.getDatabase());
+  }
+
+  public Table createTable(TableName name, Table table) {
+    return convertException(
+        () -> {
+          try {
+            return stub.createTable(
+                CreateTableRequest.newBuilder()
+                    .setParent(getDatabase(name).toString())
+                    .setTableId(name.getTable())
+                    .setTable(table)
+                    .build());
+          } catch (com.google.api.gax.rpc.AlreadyExistsException e) {
+            throw new AlreadyExistsException(e, "Table already exists: %s", name.getTable());
+          }
+        },
+        name.getTable());
+  }
+
+  public Table getTable(TableName name) {
+    if (name.getTable().isEmpty()) {
+      throw new NoSuchTableException("BigLake API does not allow tables with empty ID");
+    }
+    return convertException(
+        () -> {
+          try {
+            return stub.getTable(GetTableRequest.newBuilder().setName(name.toString()).build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table does not exist: %s (or permission denied)", name.getTable());
+          }
+        },
+        name.getTable());
+  }
+
+  public Table updateTableParameters(TableName name, Map<String, String> parameters, String etag) {
+    Table.Builder builder = Table.newBuilder().setName(name.toString()).setEtag(etag);
+    builder.getHiveOptionsBuilder().putAllParameters(parameters);
+    return convertException(
+        () -> {
+          try {
+            return stub.updateTable(
+                UpdateTableRequest.newBuilder()
+                    .setTable(builder)
+                    .setUpdateMask(FieldMask.newBuilder().addPaths("hive_options.parameters"))
+                    .build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table does not exist: %s (or permission denied)", name.getTable());
+          }
+        },
+        name.getTable());
+  }
+
+  public Table renameTable(TableName name, TableName newName) {
+    return convertException(
+        () -> {
+          try {
+            return stub.renameTable(
+                RenameTableRequest.newBuilder()
+                    .setName(name.toString())
+                    .setNewName(newName.toString())
+                    .build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table does not exist: %s (or permission denied)", name.getTable());
+          } catch (com.google.api.gax.rpc.AlreadyExistsException e) {
```

Comment:
Because this one returns "table already exists", convertException handles "namespace already exists". I can also combine them and don't distinguish table/namespace in the error message. Please let me know if you prefer this way.

Reply:
Removed convertException that takes no effect.